packages/core/src/utils/should-ignore-span.ts

```diff
@@ -0,0 +1,67 @@
+import type { ClientOptions } from '../types-hoist/options';
+import type { SpanJSON } from '../types-hoist/span';
+import { isMatchingPattern, stringMatchesSomePattern } from './string';
+
+/**
+ * Check if a span should be ignored based on the ignoreSpans configuration.
+ */
+export function shouldIgnoreSpan(
+  span: Pick<SpanJSON, 'description' | 'op'>,
+  ignoreSpans: ClientOptions['ignoreSpans'],
+): boolean {
+  if (!ignoreSpans?.length) {
+    return false;
+  }
+
+  if (!span.description) {
+    return false;
+  }
+
+  // First we check the simple string/regex patterns - if the name matches any of them, we ignore the span
+  const simplePatterns = ignoreSpans.filter(isStringOrRegExp);
+  if (simplePatterns.length && stringMatchesSomePattern(span.description, simplePatterns)) {
+    return true;
+  }
+
+  // Then we check the more complex patterns, where both parts must match
+  for (const pattern of ignoreSpans) {
+    // Have already checked for simple patterns, so we can skip these
+    if (isStringOrRegExp(pattern) || (!pattern.name && !pattern.op)) {
+      continue;
+    }
+
+    const nameMatches = pattern.name ? isMatchingPattern(span.description, pattern.name) : true;
+    const opMatches = pattern.op ? span.op && isMatchingPattern(span.op, pattern.op) : true;
+
+    if (nameMatches && opMatches) {
+      return true;
+    }
```

Comment:
this is pretty clever, took me a minute to understand :D 
maybe we can add a comment that it's only safe to return `true` here because we already bailed earlier if both op and name were undefined. 
```suggestion
    // because of the check above, only one one of them can be undefined, so falling
    // back to `true` (in combination with the check below) is safe here
    const nameMatches = pattern.name ? isMatchingPattern(span.description, pattern.name) : true;
    const opMatches = pattern.op ? span.op && isMatchingPattern(span.op, pattern.op) : true;

    if (nameMatches && opMatches) {
      return true;
    }
```